### PR TITLE
remove the callbacks onColumnRemove onColumnRename onCardRemove

### LINKED ISSUE #140: Remove the need of the callbacks
When the user removes a lane, the board calls for the `onLaneRemove` callback whether it is informed or not.
it doesn't make sense when the user remove a lane through the custom lane header template. The developer is being forced to pass a `noop` function in these cases.
It should be optional even when the board is using the default lane header template.

The same is happening to the other callbacks:  `onLaneRename`, `onCardRemove`...

The exception is to the `onLaneNew` callback, where it must return the new lane with the new id (we have plans to change this behaviour in the future).

### PR DESCRIPTION
**Pull request by deepakmountble**
* [ ] Documentation
Remove the need of the callbacks #140

### CODE REVIEW

File: src/components/Board/index.js [hunk 1]
```diff
@@ -40,15 +40,12 @@ function UncontrolledBoard({
...
   onNewColumnConfirm,
-  onColumnRemove,
   renderColumnHeader,
   allowRemoveColumn,
   allowRenameColumn,
-  onColumnRename,
   onCardNew,
   renderCard,
   allowRemoveCard,
-  onCardRemove,
```
Comment (MatheusPoliCamilo) [inline src/components/Board/index.js:51]: The onColumnRemove, onColumnRename and onCardRemove callbacks are still needed, they just shouldn't be mandatory as they are currently, forcing us to specify them even if we don't need to do anything after these actions.

I made a [sandbox](https://codesandbox.io/s/issue-140-react-kanban-23chs) to better demonstrate the problem to you.

**Review by matheus-poli — CHANGES_REQUESTED** [review]

File: src/components/Board/index.js [hunk 2]
```diff
@@ -105,11 +104,11 @@ function UncontrolledBoard({
 
   function handleCardRemove(column, card) {
     const boardWithoutCard = removeCard(board, column, card)
-    onCardRemove(
-      boardWithoutCard,
-      boardWithoutCard.columns.find(({ id }) => id === column.id),
-      card
-    )
+    { onCardRemove && onCardRemove(
```
[the remark this answers is not in the document]
Comment (deepakmountble) [reply src/components/Board/index.js]: @MatheusPoliCamilo  hey can you help me why is this error both are 2 different codes.

[the remark this answers is not in the document]
Comment (MatheusPoliCamilo) [reply src/components/Board/index.js]: Hey @deepakmountble, sorry for keep you waiting! I've made some [contributions](https://github.com/deepakmountble/react-kanban/pull/1) that maybe fix this problem to you, can you see that everything is right and accept them? Thanks!!!

**Review by matheus-poli — CHANGES_REQUESTED** [review]
You need to create Jest specs for the new behavior that you added to the project. If you need any help, please call me and I will try to help you!

**Review by lourenci — CHANGES_REQUESTED** [review]
Sorry for the late on this.

I think we have another candidate for an optional callback. Can you take a look at the `onColumnNew`?

As for the specs, I think you can use a similar existing context to complement with yours. E.g.:
```
describe('about the column renaming', () => {
  describe('when the component use the default header template', () => {
    describe('when the component receives the "allowRenameColumn" prop', () => {
      describe('when the component does not receive the "onColumnNew" prop', () => {
          it('does not call the "onColumnRename" callback when the user renames a lane', () => {

```
I've plans to refactor those contexts, don't be afraid to add your behavior. The priority here is they be tested.

Please, update the docs to make those callbacks optionals.

I write down some annotations that should ditch codeclimate issues.

Thank you for your contribution.

**Comment by Sabrinovsky** [comment]
Hey @deepakmountble still working on this PR? 
Looks like theres some requestest changes not applied yet, and also some problems with the linter.
If you need something let us know.

**Pull request closed without merging**